torch/_dynamo/variables/builtin.py

```diff
@@ -353,7 +372,25 @@ def call_function(
                 ),
                 **options,
             )
+        if any([isinstance(x, DynamicShapeVariable) for x in args]) or any(
```

Comment:
I still don't conceptually understand why you do an any test here; it doesn't feel compositional with other conditions

Reply:
Just because a function is called with a symint variable, doesn't mean the output is a symint

Reply:
Agreed, this is a hack to get a lot of enablement for cheap. It seems to be alright as most of these are builtins. We should probably call wrap_fx_proxy here. I will play with how I can improve this before I push.

Reply:
Thinking about this, are there any builtins where this is not sound?

Reply:
Comparison builtins trigger guard and return bool. If you slice a symint you should return a slice with a symint in it.

Reply:
There's a fixed set of builtins which SymInt/SymFloat work with. I'd rather just enumerate them all in the conditional.

Reply:
We had that, actually, kinda? But it didn't work as it was written and I pulled it for this PR to get tests passing. Let's either:

A) put a pause on this merge and I can go back to symbolic-shapes, disable this codepath, and re-enable-as-it-fails with explicit builtins

B) Land this as is, do A above, and then port it back to here.

I kind of prefer A...

Reply:
So the thing with having a branch is you don't have to merge everything, you can yank questionable stuff and land it later.